sdk/inc/azure/iot/az_iot_adu_ota.h

```diff
@@ -0,0 +1,284 @@
+  az_iot_adu_ota_update_manifest_instructions instructions;
+  az_iot_adu_ota_update_manifest_file files[AZ_IOT_ADU_OTA_FILE_URL_MAX_COUNT];
+  uint32_t files_count;
+  az_span create_date_time;
+} az_iot_adu_ota_update_manifest;
+
+/**
+ * @brief Verifies if the Azure Plug-and-Play writable properties component
+ *        is for ADU device update.
+ *
+ * @param[in] component_name    #az_span pointing to the component name in the
+ *                              writable properties.
+ * @return A boolean indicating if the component name is for ADU device update.
+ */
+AZ_NODISCARD bool az_iot_adu_ota_is_component_device_update(az_span component_name);
+
+/**
+ * @brief Generates the Azure Plug-and-Play (reported) properties payload
+ *        with the state of the ADU agent.
+ *
+ * @param[in] iot_hub_client        A pointer to the #az_iot_hub_client instance.
+ * @param[in] device_information    A pointer to a #az_iot_adu_ota_device_information
+ *                                  structure with all the details of the device,
+ *                                  as required by the ADU service.
+ * @param[in] agent_state           An integer value indicating the current state of
+ *                                  the ADU agent. Use the values defined by the
+ *                                  #AZ_IOT_ADU_OTA_AGENT_STATE macros in this header.
+ *                                  Please see the ADU online documentation for more
+ *                                  details.
+ * @param[in] workflow              A pointer to a #az_iot_adu_ota_workflow instance
+ *                                  indicating the current ADU workflow being processed,
+ *                                  if an ADU service workflow was received. Use NULL
+ *                                  if no device update is in progress.
+ * @param[in] last_install_result   A pointer to a #az_iot_adu_ota_install_result
+ *                                  instance with the results of the current or past
+ *                                  device update workflow, if available. Use NULL
+ *                                  if no results are available.
+ * @param[in] payload               An #az_span pointing to the memory buffer where to
+ *                                  write the resulting Azure Plug-and-Play properties.
+ * @param[out] out_payload          A pointer to hold the resulting #az_span slice of
+ *                                  `payload`, fitted to the content generated by this
+ *                                  function.
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD az_result az_iot_adu_ota_get_properties_payload(
+    az_iot_hub_client const* iot_hub_client,
+    az_iot_adu_ota_device_information* device_information,
+    int32_t agent_state,
+    az_iot_adu_ota_workflow* workflow,
+    az_iot_adu_ota_install_result* last_install_result,
+    az_span payload,
```

Comment:
Should this take an `az_span payload' or a `az_json_writer payload`?  

The scenario I'm mostly interested in is suppose the app developer has a bunch of JSON properties to respond to - some device update, others unrelated.  Assume they're using `az_iot_hub_client_properties.h` which deals with `az_iot_hub_client_properties_writer_` and is using `ref_json_writer` as the mechanism to keep data going.

Would it be easy for customer to just dump the span into the json_writer (specifically no memcpy)?

We'll probably need to discuss this out but I want to at least capture a possible optimiation and understand how it impacts the case of ADU responses only.